components/style/values/computed/transform.rs

```diff
@@ -177,3 +241,120 @@ impl TransformList {
+                Ok(GenericTransformOperation::TranslateY(
+                    t.to_animated_zero()?,
+                ))
+            },
+            GenericTransformOperation::TranslateZ(ref t) => {
+                Ok(GenericTransformOperation::TranslateZ(
+                    t.to_animated_zero()?,
+                ))
+            },
+            GenericTransformOperation::Scale3D(..) => {
+                Ok(GenericTransformOperation::Scale3D(1.0, 1.0, 1.0))
+            },
+            GenericTransformOperation::Scale(_, None) => {
+                Ok(GenericTransformOperation::Scale(1.0, None))
+            },
+            GenericTransformOperation::Scale(_, Some(_)) => {
+                Ok(GenericTransformOperation::Scale(1.0, Some(1.0)))
+            },
+            GenericTransformOperation::ScaleX(..) => {
+                Ok(GenericTransformOperation::ScaleX(1.0))
+            },
+            GenericTransformOperation::ScaleY(..) => {
+                Ok(GenericTransformOperation::ScaleY(1.0))
+            },
+            GenericTransformOperation::ScaleZ(..) => {
+                Ok(GenericTransformOperation::ScaleZ(1.0))
+            },
+            GenericTransformOperation::Rotate3D(x, y, z, a) => {
+                let (x, y, z, _) = TransformList::get_normalized_vector_and_angle(x, y, z, a);
+                Ok(GenericTransformOperation::Rotate3D(x, y, z, Angle::zero()))
+            },
+            GenericTransformOperation::RotateX(_) => {
+                Ok(GenericTransformOperation::RotateX(Angle::zero()))
+            },
+            GenericTransformOperation::RotateY(_) => {
+                Ok(GenericTransformOperation::RotateY(Angle::zero()))
+            },
+            GenericTransformOperation::RotateZ(_) => {
+                Ok(GenericTransformOperation::RotateZ(Angle::zero()))
+            },
+            GenericTransformOperation::Rotate(_) => {
+                Ok(GenericTransformOperation::Rotate(Angle::zero()))
+            },
+            GenericTransformOperation::Perspective(..) |
+            GenericTransformOperation::AccumulateMatrix { .. } |
+            GenericTransformOperation::InterpolateMatrix { .. } => {
+                // Perspective: We convert a perspective function into an equivalent
+                //     ComputedMatrix, and then decompose/interpolate/recompose these matrices.
+                // AccumulateMatrix/InterpolateMatrix: We do interpolation on
+                //     AccumulateMatrix/InterpolateMatrix by reading it as a ComputedMatrix
+                //     (with layout information), and then do matrix interpolation.
+                //
+                // Therefore, we use an identity matrix to represent the identity transform list.
+                // http://dev.w3.org/csswg/css-transforms/#identity-transform-function
+                //
+                // FIXME(nox): This does not actually work, given the impl of
```

Comment:
Why `FIXME(nox)`? We have a fallback method you can pass instead.

Reply:
Yeah, it works now.  This code was moved wholesale.